nativescript-angular/router/ns-location-strategy.ts

```diff
@@ -604,15 +607,16 @@ export class NSLocationStrategy extends LocationStrategy {
         return outlet;
     }
 
-    private updateStates(outlet: Outlet, currentSegmentGroup: UrlSegmentGroup): boolean {
+    private updateStates(outlet: Outlet, currentSegmentGroup: UrlSegmentGroup, queryParams: Params): boolean {
         const isNewPage = outlet.states.length === 0;
         const lastState = outlet.states[outlet.states.length - 1];
         const equalStateUrls = outlet.containsTopState(currentSegmentGroup.toString());
 
         const locationState: LocationState = {
             segmentGroup: currentSegmentGroup,
             isRootSegmentGroup: false,
-            isPageNavigation: isNewPage
+            isPageNavigation: isNewPage,
+            queryParams: {...queryParams}
```

Comment:
May be best to use some protection here in case queryParams comes in null or undefined?
```
queryParams: { ...(queryParams || {}) }
```